### PR TITLE
Feat stroop

### PR DESCRIPTION
**Pull request by galenwinsor**
Stroooppppp

### CODE REVIEW

File: src/config/experiment.js [hunk 1]
```diff
@@ -30,35 +29,39 @@ const getConfig = async (participantID, studyID) => {
...
  */
-const overrideSettings = (override, defaultBlockSettings) => {
-  console.log("Default block settings:", defaultBlockSettings)
-  console.log("override:", override)
-  const newBlock = deepCopy(defaultBlockSettings)
-  Object.assign(newBlock, override)
-
-  return newBlock
-}
+// const overrideSettings = (override, defaultBlockSettings) => {
```
Comment (mcmcgrath13) [inline src/config/experiment.js:50]: why aren't they used? When would they be used? provide a bit more guidance here

File: src/trials/choice.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    // A type of trial that calls a function rather than display a stimulus.
+    type: "html_keyboard_response",
+    // Allows the trial to be manually ended with the done() function.
+    response_ends_trial: true,
+    stimulus:
+      `<div class="width-100-view height-100-view">
+            <p id="color-display" class="centered-h-v font-weight-bold font-size-large" style="color:${color}">${word}</p>
+        </div>`,
+    on_start: () => {
+      removeCursor("experiment")
```
Comment (mcmcgrath13) [inline src/trials/choice.js:25]: does the cursor need to be added back?

File: src/trials/showEarnings.js [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+/**
+ * Displays the earnings from a trial.
+ * @param {number} duration How long to display the earnings.
+ * @returns trial The jsPsych trial object.
+ */
+const showEarnings = (duration) => {
+  // Maps colors to their corresponding key code.
+  const colorKeyMappings = {
+    red: 82,
```
Comment (mcmcgrath13) [inline src/trials/showEarnings.js]: should the key map be part of the config instead? if someone added `'green'` to the config, would it work?

File: src/trials/choice.js (hunk 2 above)
Comment (galenwinsor) [reply src/trials/choice.js:25]: It is in `showEarnings`

File: src/trials/showEarnings.js (hunk 3 above)
Comment (galenwinsor) [reply src/trials/showEarnings.js]: Actually fixed this in the next commit!

File: src/App.css [hunk 4]
```diff
@@ -26,6 +38,30 @@ html, body, #root, .App {
   width: 100%;
 }
 
+.width-100-view {
+  width: 100vw;
+}
+
+.width-min-content {
+  width: max-content;
```
Comment (mcmcgrath13) [inline src/App.css:46]: class and css don't really match

File: src/timelines/preamble.js [hunk 5]
```diff
@@ -1,25 +1,54 @@
...
-if (envConfig.USE_PHOTODIODE) {
-  timeline.push(holdUpMarker())
-  timeline.push(startCode())
-}
+const preamble = (experimentConfig) => {
+  const colors = experimentConfig.conditions
+  let colorList = ""
+
+  for (let i = 0; i < colors.length; i++) {
+    colorList += `<li class="font-size-large"><span style="color: ${colors[i]}">WORD</span>: ${colors[i][0].toUpperCase()}</li>`
```
Comment (mcmcgrath13) [inline src/timelines/preamble.js]: isn't the color list a flat array, how does the taking the `[0]` index work?

Comment (mcmcgrath13) [reply src/timelines/preamble.js]: oh, I see - it's the first key - add a comment

File: src/timelines/taskBlock.js [hunk 6]
```diff
@@ -5,14 +5,8 @@ const taskBlock = (blockSettings) => {
   // initialize block
 	const startingOpts = generateStartingOpts(blockSettings)
 
-  const blockDetails = {
-	  block_earnings: 0.0,
-		optimal_earnings: 0.0,
-		continue_block: true
-	}
-
 	// timeline = loop through trials
```
Comment (mcmcgrath13) [inline src/timelines/taskBlock.js:8]: why not collect the overall earnings anymore?

File: src/trials/choice.js (hunk 2 above)
Comment (mcmcgrath13) [reply src/trials/choice.js:25]: add a comment on that

File: src/trials/showEarnings.js [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+      } else if (keyPressed === lastColor[0].toUpperCase()) {
+        trial.stimulus = earningsDisplay(1)
+      } else {
+        trial.stimulus = earningsDisplay(-1)
+      }
+      if (envConfig.USE_PHOTODIODE) trial.stimulus += photodiodeGhostBox();
+    },
+    on_finish: (data) => {
+      data.code = eventCodes.show_earnings
+      addCursor("experiment");
```
Comment (mcmcgrath13) [inline src/trials/showEarnings.js]: add a comment on where this was removed

File: src/App.css (hunk 4 above)
Comment (galenwinsor) [reply src/App.css:46]: Yup, oops!

File: src/timelines/taskBlock.js (hunk 6 above)
Comment (galenwinsor) [reply src/timelines/taskBlock.js:8]: I don't think it was ever used anywhere.

File: src/trials/choice.js (hunk 2 above)
Comment (mcmcgrath13) [reply src/trials/choice.js:25]: instead of commenting on what's happening, comment on when we add the cursor back

**Review by mcmcgrath13 — CHANGES_REQUESTED** [review]

**Review by mcmcgrath13 — APPROVED** [review]
🔴 🟦 💛

**Pull request merged**